[jwsung91-unilink] Add new port

jwsung91 opened the pull request:
Adds the **unilink** communication library as a new vcpkg port: **jwsung91-unilink**.

**unilink** is a cross-platform asynchronous communication library built on Boost.Asio,
providing a unified interface over TCP, Serial, and other transports.

The port name follows the `githuborg-reponame` convention to avoid ambiguity and
potential name collisions, as discussed during review.

## Usage example

```cmake
find_package(unilink CONFIG REQUIRED)

add_executable(main main.cpp)
target_link_libraries(main PRIVATE unilink::unilink_shared)
# or:
# target_link_libraries(main PRIVATE unilink::unilink_static)
```

## New Port Checklist

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

jwsung91 committed:
[unilink] Add port for v0.1.3

jwsung91 commented:
@microsoft-github-policy-service agree

jwsung91 commented:
This failure seems unrelated to the `unilink` port itself.

All failing jobs report:

`Path does not exist: .../scripts/list_files`

This file is part of the CI infrastructure, not part of the port.  
Linux builds succeed and local builds with `./vcpkg install unilink:x64-linux` also succeed.

Could you please confirm if this is a CI issue?

jwsung91 committed:
[unilink] change version to v0.1.4

jwsung91 committed:
[unilink] change version to v0.1.5

jwsung91 committed:
[unilink] change version to v0.1.6

jwsung91 committed:
fix: remove older versions

jwsung91 committed:
fix: apply suggestion @vicroms

jwsung91 committed:
fix: apply suggestion @BillyONeal

jwsung91 committed:
[unilink] change version to v0.1.7

jwsung91 committed:
feat: update unilink to version 0.1.8

jwsung91 committed:
unilink: fix sha512

jwsung91 committed:
[unilink] change version to v0.1.9

jwsung91 committed:
[unilink] fix version v0.1.9 git-tree

jwsung91 committed:
[unilink] remove support for uwp

jwsung91 committed:
[unilink] fix static issue

jwsung91 committed:
[unilink] fix: remove build options

jwsung91 committed:
[unilink] fix: change git-tree

jwsung91 commented:
All previously discussed issues have been addressed.

- UWP is now marked as unsupported via the manifest (`"supports": "!uwp"`).
- The portfile now follows `VCPKG_LIBRARY_LINKAGE` when configuring CMake, so shared vs. static builds match the selected triplet.

CI has been rerun and is now passing across the supported triplets.
Could you please take another look when you have a moment?

dg0yt commented on ports/unilink/portfile.cmake:
```diff
@@ -0,0 +1,33 @@
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS
+        -DBUILD_SHARED_LIBS=${UNILINK_BUILD_SHARED}
```
```suggestion
```
This is already set by `vcpkg_configure_make`.

dg0yt commented on ports/unilink/portfile.cmake:
```diff
@@ -0,0 +1,33 @@
+    REF v0.1.9
+    SHA512 0c159181c2fc5ca3ec0dd913770ab771f659ccf71e16cb3ddb84030c37e24ce187f0141ac073d8c1514c20795f6da145ecd7a14663711b0a9ce3edf178a1251c)
+
+# Follow the triplet linkage (dynamic/shared vs static)
```
```suggestion
```

dg0yt reviewed (commented):
LGTM but somebody else must approve.
A few nitpicks.

jwsung91 committed:
Update ports/unilink/portfile.cmake

remove comment

Co-authored-by: Kai Pastor <dg0yt@darc.de>

jwsung91 committed:
Update ports/unilink/portfile.cmake

apply install copyright

Co-authored-by: Kai Pastor <dg0yt@darc.de>

jwsung91 committed:
Update ports/unilink/portfile.cmake

remove already set by `vcpkg_configure_make`

Co-authored-by: Kai Pastor <dg0yt@darc.de>

jwsung91 commented:
@dg0yt  
Thanks for the suggestions — they all made sense, so I’ve applied them.
This PR review has been very helpful.

jwsung91 committed:
[unilink] chore: update git-tree

BillyONeal reviewed (changes requested):
This port is great but I think the name of the port needs to follow the githuborg-reponame form unless I misunderstood

jwsung91 commented:
I’m currently at work, but I’ll make the rename and push the updates right after work.

Thanks for your patience.

jwsung91 committed:
[unilink] fix: rename port to jwsung91-unilink (@BillyONeal)

jwsung91 commented:
I’ve renamed the port to `jwsung91-unilink` and updated the port files, version database, and usage documentation accordingly.

Thanks for your time and review!

BillyONeal reviewed (approved):
Thanks for the new port!
